src/main/java/com/fasterxml/jackson/databind/util/TokenBuffer.java

```diff
@@ -276,13 +276,23 @@ public JsonParser asParser(ObjectCodec codec)
         return new Parser(_first, codec, _hasNativeTypeIds, _hasNativeObjectIds, _parentContext);
     }
 
+    /**
+     * @param streamReadConstraints constraints for streaming reads
+     * @since v2.15
+     */
+    public JsonParser asParser(StreamReadConstraints streamReadConstraints)
+    {
+        return new Parser(_first, _objectCodec, _hasNativeTypeIds, _hasNativeObjectIds, _parentContext, streamReadConstraints);
```

Comment:
there are still calls to variants of `asParser(...)` that don't pass in StreamReadConstraints and this will probably still lead to issues - because StreamReadConstraints.defaults() applies a 1000 char limit on numbers

Reply:
Ok thank you -- I had thought this will probably become an issue soon. Should have just created an issue proactively and cc'd you.